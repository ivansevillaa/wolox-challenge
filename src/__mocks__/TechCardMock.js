const TechCardMock = {
  tech: 'Node',
  year: '2009',
  author: 'Ryan Dahl',
  license: 'MIT',
  language: 'JavaScript',
  type: 'Back-End',
  logo:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Node.js_logo.svg/220px-Node.js_logo.svg.png',
};

export default TechCardMock;
