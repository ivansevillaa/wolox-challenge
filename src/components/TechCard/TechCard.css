.tech-card {
  height: 50rem;

  background-color: var(--color-light);
  border-radius: 10px;
  border: 1px solid var(--color-gray);
  box-shadow: 0px 5px 5px 0px rgba(229, 229, 229, 0.75);

  transition: 500ms;
}

.tech-card:hover {
  transform: scale(1.1);
  transform: translateY(-10px);
}

.tech-card__image {
  width: 100%;
  height: 40%;

  border-top-left-radius: inherit;
  border-top-right-radius: inherit;
  border-bottom: inherit;
  object-fit: contain;
}

.tech-card__details {
  padding: var(--spacing-0) var(--spacing-4);
}

.tech-card__details > h3 {
  color: var(--color-dark);
  font-size: var(--font-size-normal);
}

.tech-card__details > p {
  color: var(--color-dark);
}

.tech-card__details > p > span {
  font-weight: var(--font-weight-medium);
}
