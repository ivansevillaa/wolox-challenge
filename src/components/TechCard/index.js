import React from 'react';
import PropTypes from 'prop-types';
import './TechCard.css';

const TechCard = ({ data }) => (
  <article className="tech-card">
    <img
      className="tech-card__image"
      src={data.logo}
      alt={`${data.tech} logo`}
    />
    <div className="tech-card__details">
      <h3>{data.tech}</h3>
      <p>
        <span>Language:</span> {data.language}
      </p>
      <p>
        <span>Year:</span> {data.year}
      </p>
      <p>
        <span>License:</span> {data.license}
      </p>
      <p>
        <span>Author:</span> {data.author}
      </p>
      <p>
        <span>Type:</span> {data.type}
      </p>
    </div>
  </article>
);

TechCard.propTypes = {
  data: PropTypes.objectOf(PropTypes.string),
};

export default TechCard;
