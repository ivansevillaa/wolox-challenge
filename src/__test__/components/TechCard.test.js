import React from 'react';
import { render, screen } from '@testing-library/react';
import TechCard from '../../components/TechCard';
import TechCardMock from '../../__mocks__/TechCardMock';

describe('TechCard', () => {
  test('render the correct tech name', () => {
    render(<TechCard data={TechCardMock} />);

    expect(screen.getByRole('heading').textContent).toMatch(/Node/);
  });

  test('TechCard snapshot', () => {
    const { asFragment } = render(<TechCard data={TechCardMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
