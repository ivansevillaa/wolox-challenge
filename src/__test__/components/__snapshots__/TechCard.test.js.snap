// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechCard TechCard snapshot 1`] = `
<DocumentFragment>
  <article
    class="tech-card"
  >
    <img
      alt="Node logo"
      class="tech-card__image"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Node.js_logo.svg/220px-Node.js_logo.svg.png"
    />
    <div
      class="tech-card__details"
    >
      <h3>
        Node
      </h3>
      <p>
        <span>
          Language:
        </span>
         JavaScript
      </p>
      <p>
        <span>
          Year:
        </span>
         2009
      </p>
      <p>
        <span>
          License:
        </span>
         MIT
      </p>
      <p>
        <span>
          Author:
        </span>
         Ryan Dahl
      </p>
      <p>
        <span>
          Type:
        </span>
         Back-End
      </p>
    </div>
  </article>
</DocumentFragment>
`;
